File: tests/cases/enforce/sealed/python/inputs/src/orders/annotated.py
from orders.receipt import Receipt


class AnnotatedReceipt:
    """Composition instead of inheritance: holds a Receipt, does not extend
    it. This is the sanctioned way to add behaviour to a sealed type."""

    def __init__(self, receipt: Receipt, note: str) -> None:
        self.receipt: Receipt = receipt
        self.note: str = note

File: tests/cases/enforce/sealed/lua/inputs/src/orders/annotated.lua
local Receipt = require("orders.receipt")

-- Composition instead of inheritance: holds a Receipt, does not derive from
-- it. `Receipt.new` inside a constructor is construction, not subtyping.
local Annotated = {}
Annotated.__index = Annotated

function Annotated.new(receipt, note)
  local self = setmetatable({}, Annotated)
  self.receipt = receipt
  self.note = note
  return self
end

return Annotated
